### PR TITLE
Vpc creation optional

### PR DESCRIPTION
**Pull request by woodcockr**
# Why this change is needed
> Describe why this change is needed, what issues it will fix and the benefits the change will add

Some deployments have IT requirements that necessitate the use of an alternate VPC configuration to the default created by the `odc_eks` module. This PR makes VPC creation optional and allows for the VPC and subnets to be supplied to the `odc_eks` module.

# Negative effects of this change
> Will making this change break or change an existing functionality? flag it here

Making `odc_eks.module.vpc` optional adds a `count` and the resource path for the default `create_vpc = true` configuraiton changes to `odc_eks.module.vpc[0]`. It is thus necessary to `terrafrom state mv` all of these resources to the new path. The new `CHANGELOG.md` contains the necessary commands.
Once the resource names are updated there is _no change to infrastructure_ on `terraform plan`.

Since the state mv is required but the outcome is no change it is proposed the version tag for this release be a minor version increment to `v1.10.0` as reflected in the CHANGELOG.md

Note:
  * If the terraform state is not moved to the new path prior to terraform plan then the plan will fail with the kuberenetes provider and resources throwing errors about not being able to connect to the kubernetes cluster (which it can't because the vpc doesn't exist in the new resource path)
  * Terraform outputs will show as being updated in the terraform plan but the output is identical. This appears to be an internal change due to the new module path for the output but has no impact downstream as the value is identical.

### CODE REVIEW

**Comment by NikitaGandhi** [comment]
adding @mmochan and @lbodor to review this PR as this change will impact their EKS setup too.

**Comment by NikitaGandhi** [comment]
@woodcockr @mmochan I am ok with this change and version `v1.10.0` release but to minimise an impact for our DEA EKS internal setup, let's do following:

- cut one more release from master branch `v1.9.5` (as master has few minor updates beyond `v1.9.4`)
- point modules to released version instead of `master` branch
- perform version increment to `v1.10.0` as describe in the CHANGELOG.md to catchup with latest

Let me know your thoughts.

**Comment by woodcockr** [comment]
@NikitaGandhi Your pointing live deployments at master branch! You are braver than I am ;-)

**Comment by mmochan** [comment]
@NikitaGandhi I'm good with that approach.

**Review by lbodor — APPROVED** [review]
This will work for us. I've updated our terraform state in dev, which did result in a no-op plan. Thanks very much for checking in with us.

**Comment by woodcockr** [comment]
Apparently I can't merge this despite @lbodor approval because it needs to be reviewed by someone with write access.
@NikitaGandhi could you please approve?

**Review by NikitaGandhi — APPROVED** [review]

File: CHANGELOG.md
Comment (lbodor) [inline CHANGELOG.md:52]: `terraform state mv module.odc_eks.module.vpc module.odc_eks.module.vpc[0]`

Comment (woodcockr) [reply CHANGELOG.md:53]: Thanks, missing one added.

Comment (lbodor) [reply CHANGELOG.md:53]: I meant that that one is the only one you need.

**Pull request merged**
